CDCgov/prime-simplereport
# Add ability to dismiss toast notifications (#4058)

# FRONTEND PULL REQUEST

## Related Issue
Resolves #4058 

## Changes Proposed
- add an "X" button to the toast notifications so a user can close them without having to wait for them to auto close
- update the designs to match the [figma](https://www.figma.com/file/NPSPM5SBzQ7Ra3mmr9M9cB/SimpleReport-Public-Site?node-id=12545%3A34043)
- create a `<SRToastContainer>` component that has all the defaults for `<ToastContainer>`
- create a `showSuccess` and use `showError` to display toast notifications to avoid having to passing `<Alert>`s 

## Additional Information
- Noticed some patterns in the support admin pages where we were showing toast notifications but a user would never actually see this because they are immediately redirected (e.g. https://github.com/CDCgov/prime-simplereport/pull/4451#discussion_r995092415) If we want to fix this up in a separate story, please let me know and I can create an issue for it.
- Kenny noticed while archiving a test, that the toaster message appears right before the screen reloads. the toaster should appear after the screen reloads to give the user time to view the message.
- Considering the two bullet points above, we may want a story to check/clean up the toast message display experience throughout the app.
- As seen in the comments for the original issue linked above, Kenny decided and asked that the toast notification be moved to the bottom left hand corner and to keep the auto close at 5 seconds.

## Testing
- Go to all different parts of the app that generate a success or error message (e.g. Results tab, /sign-up, Manage organization form, patient bulk upload with errors etc...)
- Ensure the change has not introduced any new accessibility concerns

## Screenshots / Demos
<img width="579" alt="Screen Shot 2022-10-13 at 19 55 20" src="https://user-images.githubusercontent.com/20211771/195738053-29874e7f-caf9-4d73-8ec6-1368f3260cab.png">

<img width="704" alt="Screen Shot 2022-10-13 at 19 54 58" src="https://user-images.githubusercontent.com/20211771/195738061-59fe2997-c643-44b0-9610-9ec630deecdb.png">
